backend/question-service/src/controllers/handlers/get-handler.ts

```diff
@@ -1,42 +1,60 @@
 import { Request, Response } from "express";
 import HttpStatusCode from "../../lib/enums/HttpStatusCode";
 import { QueryParamValidator } from "../../lib/validators/QueryParamValidator";
-import questionDb from "../../models/database/schema/question";
-import mongoose from "mongoose";
+import db from "../../models/db";
 import { ZodError } from "zod";
 import { formatErrorMessage } from "../../lib/utils/errorUtils";
 import Topic from "../../lib/enums/Topic";
+import Complexity from "../../lib/enums/Complexity";
+import { Example } from "../../models/question";
 
 // Check if database connection is successful
 export const getHealth = async (_: Request, response: Response) => {
-  if (mongoose.connection.readyState === 1) {
+  try {
+    const result = await db.$queryRaw`SELECT 1`;
+
+    if (!result) {
+      throw new Error("No database connection from the server");
+    }
+
     response.status(HttpStatusCode.OK).json({ message: "Healthy" });
-  } else {
+  } catch (error) {
+    console.log(error);
     response.status(HttpStatusCode.INTERNAL_SERVER_ERROR).json({
-      message: "Database connection is not established.",
+      error: "INTERNAL SERVER ERROR",
+      message: "No database connection from the server",
     });
   }
 };
 
+type Filter = {
+  topics?: string[];
+  difficulties?: string[];
+  author?: string;
+};
+
 // Query mulitple questions from database by filter
 export const getQuestions = async (request: Request, response: Response) => {
   try {
     // check if there are any query params provided and the provided values are valid
-    const { topics, complexity, author } = QueryParamValidator.parse(
+    const { topics, difficulties, author } = QueryParamValidator.parse(
```

Comment:
Hmm... possible to standardize this? seems like question pages use complexity while user preference named it difficulties...

Reply:
Oh, did you use these query parameters in `getQuestions` API call so far? I was trying to standardise with user `preferences` as it is set as `difficulties` too. :P
I was just trying to avoid using the word `complexities` because it sounds weird, but if you all prefer using `complexities` instead, I can change to that. By the way, this `complexities` is either a string, a string array, or undefined fyi.

Reply:
Apologise for the confusion here. After a second look I realised that I should just named all the query params here as a singular noun, eg: just `topic`, `complexity`. Because when we used it, we did something like `topic=String&topic=Two Pointers&complexity=Easy&complexity=Hard`, so there is no need to name them in plural form.